src/estimagic/optimization/tiktak.py

```diff
@@ -456,17 +461,38 @@ def update_convergence_state(current_state, starts, results, convergence_criteri
     best_y = current_state["best_y"]
     best_res = current_state["best_res"]
 
+    # get indices of local optimizations that did not fail
     valid_indices = [i for i, res in enumerate(results) if not isinstance(res, str)]
 
+    # If all local optimizations failed, return early so we don't have to worry about
+    # index errors later.
     if not valid_indices:
         return current_state, False
 
+    # ==================================================================================
+    # reduce eveything to valid optimizations
+    # ==================================================================================
     valid_results = [results[i] for i in valid_indices]
     valid_starts = [starts[i] for i in valid_indices]
-
     valid_new_x = [res["solution_x"] for res in valid_results]
-    valid_new_y = [res["solution_criterion"] for res in valid_results]
+    valid_new_y = []
+
+    # make the criterion output scalar if a least squares optimizer returns an
+    # array as solution_criterion.
+    for res in valid_results:
+        if np.isscalar(res["solution_criterion"]):
+            valid_new_y.append(res["solution_criterion"])
+        else:
+            valid_new_y.append(
+                aggregate_func_output_to_value(
+                    f_eval=res["solution_criterion"],
+                    primary_key=primary_key,
+                )
+            )
```

Comment:
Why do we need to if-clause? Doesn't `aggregate_func_output_to_value` handle the scalar case as well?

Reply:
No, it does not. Usually this function is called, when we exactly know what we will get (e.g. on the function value that gets passed to a least squares optimizer). Here we are not sure how much post-processing has already been done in by the internal least squares optimizer.